### PR TITLE
feat: implement compare

### PR DESCRIPTION
**Pull request by theseion**

### CODE REVIEW

File: cmd/compare.go [hunk 1]
```diff
@@ -66,3 +82,137 @@ func rebuildCompareCommand() {
 	compareCmd = createCompareCommand()
 	buildCompareCommand()
 }
+
+// FIXME: duplicated in update.go
```
Comment (fzipi) [inline cmd/compare.go:86]: Is this broken here? Or just duplicated?

File: cmd/compare.go [hunk 2]
```diff
@@ -66,3 +82,137 @@ func rebuildCompareCommand() {
...
+				counter = "~ " + counter
+			}
+			fmt.Println(strings.Repeat(" ", 59-len(generatedChunk)), counter)
+		}
+		if printFirstDiff {
+			fmt.Println("===========")
+		}
+	}
+	fmt.Printf("\n")
+}
```
Comment (fzipi) [inline cmd/compare.go:218]: Why not https://github.com/terrastruct/diff or similar?

**Review by fzipi — COMMENTED** [review]
Left comments.

File: cmd/compare.go (hunk 1 above)
Comment (theseion) [reply cmd/compare.go:86]: Just duplicated. Should be consolidated somehow.

File: cmd/compare.go (hunk 2 above)
Comment (theseion) [reply cmd/compare.go:218]: Hm... Let me look into that.

Comment (theseion) [reply cmd/compare.go:218]: NO!
I TALKA WITTA MY HANDSA!

Comment (fzipi) [reply cmd/compare.go:218]: Ah, then I SHATAPA WITH MY SHATAPA.

**Review by fzipi — APPROVED** [review]
🚀

**Pull request merged**